cmd/scaffold.go

```diff
@@ -369,6 +369,13 @@ func (fnb *FlowNodeBuilder) initProfiler() {
 }
 
 func (fnb *FlowNodeBuilder) initDB() {
+
+	// if a db has been passed in, use that instead of creating one
+	if fnb.BaseConfig.db != nil {
+		fnb.DB = fnb.BaseConfig.db
+		return
+	}
+
 	// Pre-create DB path (Badger creates only one-level dirs)
 	err := os.MkdirAll(fnb.BaseConfig.datadir, 0700)
 	fnb.MustNot(err).Str("dir", fnb.BaseConfig.datadir).Msg("could not create datadir")
```

Comment:
It would be good if we reject the `datadir` flag when DB is passed in. Ie. make sure its not set, to avoid dangling unused configs and potential confusion

Reply:
made that change - to ignore datadir if db is passed in.